### PR TITLE
Devex 1187: CaseDeadline

### PR DESCRIPTION
**Pull request by rachelschneiderman**
This PR is part 1 of many for the card: [`Convert validation messages for entities to using Joi syntax.`](https://trello.com/c/J4S7ZhKx/1187-convert-validation-messages-for-entities-to-using-joi-syntax). It migrates the CaseDeadline entity to using default Joi error message syntax and adds a new abstract class to handle that change so we don't have to deliver all the entities migrated over at once and can instead move them one at a time to the new syntax.

### CODE REVIEW

File: shared/src/business/entities/CaseDeadline.test.ts [hunk 1]
```diff
@@ -28,54 +61,46 @@ describe('CaseDeadline', () => {
...
-You can whip your Porsche
-I been in the valley
-You ain't been up off that porch, now`,
+            Horse tack is attached
+            Hat is matte black
+            Got the boots that's black to match
+            Ridin' on a horse, ha
+            You can whip your Porsche
+            I been in the valley
+            You ain't been up off that porch, now`, // Description can be at most 120 characters
```
Comment (TomElliottFlexion) [inline shared/src/business/entities/CaseDeadline.test.ts:95]: Thank you for not changing this important part of our codebase. The lore remains intact.

**Review by TomElliottFlexion — APPROVED** [review]
Nice job keeping this small

File: shared/src/business/entities/CaseDeadline.test.ts (hunk 1 above)
Comment (rachelschneiderman) [reply shared/src/business/entities/CaseDeadline.test.ts:95]: nod

File: shared/src/business/entities/joiValidationEntity/JoiValidationEntity_New.ts
Comment (jimlerza) [inline shared/src/business/entities/joiValidationEntity/JoiValidationEntity_New.ts:1]: Will this name be changed in the final PR?

**Review by jimlerza — APPROVED** [review]

File: shared/src/business/entities/joiValidationEntity/JoiValidationEntity_New.ts
Comment (rachelschneiderman) [reply shared/src/business/entities/joiValidationEntity/JoiValidationEntity_New.ts:1]: Yes! Once we have finished moving over all the entities we will delete what is now `JoiValidationEntity.ts` and rename what is now `JoiValidationEntity_New` to remove '_New'.

**Pull request merged**